api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -240,15 +244,13 @@ public DataSet apply(DbDataset dataSet) {
         }
       };
 
-  private ConceptSet toClientConceptSet(DbConceptSet conceptSet) {
-    ConceptSet result = ConceptSetsController.toClientConceptSet(conceptSet);
-    if (!conceptSet.getConceptIds().isEmpty()) {
-      Iterable<DbConcept> concepts = conceptDao.findAll(conceptSet.getConceptIds());
-      result.setConcepts(
-          StreamSupport.stream(concepts.spliterator(), false)
-              .map(ConceptsController::toClientConcept)
-              .collect(Collectors.toList()));
-    }
+  private ConceptSet toClientConceptSet(DbConceptSet dbConceptSet) {
```

Comment:
Why not include this function in. your ConceptSEtMapper?

Reply:
^

I think it should be possible but I'd be OK with waiting until there is a mapper for `Concept` as well.

I think with that addition, it would be a lot easier for MapStruct to infer the `Long -> Concept` mapping pattern.
I'm just not sure if it will automatically pick figure out the transitive relationship of `Long -> DbConcept -> Concept` if it knows how to do `Long -> DbConcept` and `DbConcept -> Concept`.

As is, I believe you'd have to use an `AfterMapping` annotation since the static method of `ConceptsController.toClientConcept` can't be provided through DI.

Reply:
Transitive, invisible conversion chains scare me a bit. This is why C++ has `explicit` constructors. The rare cases where it runs away from you are scary. Much less scary with less common types.